woocommerce/woocommerce-gutenberg-products-block
# Followup: switch namespace on useStoreProducts hook.

Closes: #1097

In this pull:

- ~added some missing data to the `/wc/store/products` route (and updated tests) - will need verification to ensure we're not exposing something on the public route that we shouldn't.  I basically added anything the `ProductButton` was currently using from the old route.~ (done in #1116)
- `useStoreProducts` hooks has been switched to use the `/wc/store` namespace instead of `/wc/blocks`.  Now the `All Products` block can be viewed and interacted with on the frontend without being authed!
- `ProductButton` has been refactored to be a functional component and use hooks.
- As a part of the above refactor I created a custom hook internal only to the component.  I held off exporting this until we get a better idea of what pattern we'll use for cart related hooks.  The logic is still encapsulated in the hook though so it should be relatively trivial to either extract it or refactor it down the road.

## To test

- verify the All products block works on the frontend as expected _while logged out of the site_.
- verify that adding to the cart works and the button for the product is updated as expected.

### Potential things to watch for.

- in initial tests while logged out, I started getting a `{"code":"rest_cookie_invalid_nonce","message":"Cookie nonce is invalid","data":{"status":403}}` when trying to add another quantity of a product to the store.  However, I _think_ it was a cookie persistence glitch in my browser when I switched from being logged in to logged out as a hard refresh seemed to fix it.  Still, if it happens frequently enough, we may need to do a deeper look.

## Next steps

Would probably be good to do another beta build after this gets merged so that testers can verify non-authed use works as expected.
